utility_scripts/list_roles.py

```diff
@@ -0,0 +1,114 @@
+#   Copyright 2017 Massachusetts Open Cloud
+#
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+"""List all roles held by a user or in a project
+
+Script with 2 modes, depending on which of the 2 required, mutually
+exclusive options are passed:
+
+    --user <USERNAME>
+      List all projects the user belongs to and their role(s) in the project
+or
+    --project <PROJECT>
+      List all users belonging to this project and their roles
+
+Optional flag --config/-c specifies a configuration file; if none specifie,
+the script will look for one in /etc/settings.ini and ./settings.ini
+
+Usage:
+    python search_ips.py [--user <USERNAME>] [--project <PROJECT>]
+                         [-c/--config <CONFIG_FILE>]
+"""
+import argparse
+import ConfigParser
+from keystoneclient.v3 import client
+from keystoneauth1 import session
+from keystoneauth1.identity import v3
+
+# This is a hack for now until directory structure is sorted
+# When this is removed remember to un-ignore E402 in flake8
+import os
+import sys
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+PROJECT_DIR = os.path.abspath(os.path.join(THIS_DIR, os.pardir))
+sys.path.insert(0, PROJECT_DIR)
+from config import set_config_file
+
+parser = argparse.ArgumentParser(
+    description=("List all roles of the specified user in any project, or "
+                 "all users of the specified project and their roles."))
+parser.add_argument('-c', '--config',
+                    help='Specify configuration file.')
+
+mode = parser.add_mutually_exclusive_group(required=True)
+mode.add_argument('--user',
+                  help='List roles of the specified user.')
+mode.add_argument('--project',
+                  help=('List all users of the specified project and their'
+                        'roles'))
+
+args = parser.parse_args()
+
+CONFIG_FILE = set_config_file(args.config)
+
+config = ConfigParser.ConfigParser()
+config.read(CONFIG_FILE)
+
+admin_user = config.get('auth', 'admin_user')
+admin_pwd = config.get('auth', 'admin_pwd')
+admin_project = config.get('auth', 'admin_project')
+auth_url = config.get('auth', 'auth_url')
+nova_version = config.get('nova', 'version')
+
+auth = v3.Password(auth_url=auth_url,
+                   username=admin_user,
+                   user_domain_id='default',
```

Comment:
·default` is good for now. But you might want to have a global variable that is called `domain`. So later on, if we support different domains, you could just change that global variable. (even though I have no idea whether we are going to support different domains at all or not?)